src/hooks/useCalculate.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useRef, useState } from 'react';
+import { operations } from '../constants';
+import { calculate } from '../utils';
+
+const useCalculate = () => {
+  const [total, setTotal] = useState<number | null>(null);
+
+  const operatorRef = useRef<(next: number) => number>();
+  const inputCountRef = useRef<number>(0);
+
+  const handleOperationClick = (operation: keyof typeof operations) => {
+    if (total === null) return;
+
+    if (inputCountRef.current === 0) {
+      operatorRef.current = calculate[operation](total);
+      return;
+    }
+
+    const nextTotalValue = operatorRef.current
+      ? Math.floor(operatorRef.current(total))
+      : total;
+
+    setTotal(nextTotalValue);
+    operatorRef.current = calculate[operation](nextTotalValue);
+
+    inputCountRef.current = 0;
+  };
+
+  const handleDigitClick = useCallback((digit: number) => {
+    const count = inputCountRef.current;
+
+    setTotal((prevTotal) => {
+      if (!prevTotal) return digit;
+
+      if (count === 0) {
+        return digit;
+      }
+      if (count >= 3) return prevTotal;
+      return prevTotal * 10 + digit;
+    });
+
+    inputCountRef.current = count + 1;
+  }, []);
+
+  const handleClearClick = useCallback(() => {
+    setTotal(null);
+    operatorRef.current = undefined;
+    inputCountRef.current = 0;
+  }, [setTotal]);
+
+  return {
+    total,
+    handleDigitClick,
+    handleOperationClick,
+    handleClearClick,
```

Comment:
```suggestion
    insertDigit,
    operate,
    clear,
```
(마이너) 저는 반환값으로 해당 훅이 특정 요소에 종속되지 않고 다른 요소들에서도 사용해도 어색하지 않게 네이밍해주는걸 좋아해요

Reply:
[ddb12a5](https://github.com/next-step/react-calculator/pull/4/commits/ddb12a57b50308bb8edadde068445e8ff5a36dbe) 좋은 관점 같습니다 항상 이벤트 핸들러에 쓰이는 함수들은 아닐 수도 있으니깐...